### ISSUE TITLE
Keras Progress bar - new line every iteration (broken from notebook -> lab)

### ISSUE THREAD

**Issue by JonasHarnau** [jupyterlab/jupyterlab#2982]
Hi,
Moving from jupyter notebook to jupyter lab breaks the keras progress bar for me. In lab it prints a new line for every update. Minimum (non)-working example:
```
from keras.utils.generic_utils import Progbar
progbar = Progbar(10)
for i in range(10):
    progbar.update(i, force=True)
```

## notebook
![notebook](https://user-images.githubusercontent.com/25103918/30276880-240c329c-96fe-11e7-96ab-414704fccd34.png)

## lab
![lab](https://user-images.githubusercontent.com/25103918/30276866-183b5204-96fe-11e7-99dd-1f45402eb4ae.png)

**Comment by jasongrout** [comment]
Thanks for the report. It looks like this is a duplicate of https://github.com/jupyterlab/jupyterlab/issues/1104 - closing so we can shift the discussion over there.

**Comment by kast26** [comment]
I had the same problem. The following worked it for me. I changed a few lines in generic_utils.py of the keras source code (on my computer in [...]\Miniconda\envs\tensf\Lib\site-packages\keras):

Lines 311-314 of generic_utils.py were changed from:

        self._dynamic_display = ((hasattr(sys.stdout, 'isatty') and
                                  sys.stdout.isatty()) or
                                 'ipykernel' in sys.modules)
to:
        #self._dynamic_display = ((hasattr(sys.stdout, 'isatty') and
        #                          sys.stdout.isatty()) or
        #                         'ipykernel' in sys.modules)
        self._dynamic_display = True   # inserted to overwrite the above (workaround by KS)

In other words, I commented out the assignment of the self._dynamic_display variable, and instead set the variable manually to "True", so that the following code (which is elsewhere in generic_utils.py) could be executed:

            if self._dynamic_display:
                sys.stdout.write('\b' * prev_total_width)
                sys.stdout.write('\r')
 
Then the progress bar in keras worked nicely :-)
